fvm/errors/errors.go

```diff
@@ -53,19 +50,17 @@ func findImportantCodedError(err error) (CodedError, bool) {
 		return nil, true
 	}
 
-	if coded.Code().IsFailure() {
-		return coded, false
-	}
-
-	shallowest := coded
 	for {
-		if !As(coded.Unwrap(), &coded) {
-			return shallowest, false
+		if coded.Code().IsFailure() {
+			return coded, false
 		}
 
-		if coded.Code().IsFailure() {
+		var nextCoded CodedError
+		if !As(coded.Unwrap(), &nextCoded) {
 			return coded, false
 		}
+
+		coded = nextCoded
```

Comment:
how does the value of nextCoded get updated? shouldn't this be `coded = coded.Unwrap()`?

Reply:
nextCoded is updated by `As(coded.Unwrap(), &nextCoded)`.  The error is already unwrapped to the next nested coded error